unittests/src/main/java/uk/ac/manchester/tornado/unittests/vectortypes/TestInts.java

```diff
@@ -188,4 +192,106 @@ public void testDotProductDouble() {
 
         assertEquals(seqReduce[0], outputReduce[0]);
     }
+
+    public static void addVectorInt2(VectorInt2 a, VectorInt2 b, VectorInt2 results) {
+        for (@Parallel int i = 0; i < a.getLength(); i++) {
+            results.set(i, Int2.add(a.get(i), b.get(i)));
+        }
+    }
+
+    @Test
+    public void testVectorFloat2() {
+        int size = 16;
+
+        VectorInt2 a = new VectorInt2(size);
+        VectorInt2 b = new VectorInt2(size);
+        VectorInt2 output = new VectorInt2(size);
+
+        for (int i = 0; i < size; i++) {
+            a.set(i, new Int2(i, i));
+            b.set(i, new Int2(size - i, size - i));
+        }
+
+        //@formatter:off
+        new TaskSchedule("s0")
+                .task("t0", TestInts::addVectorInt2, a, b, output)
+                .streamOut(output)
+                .execute();
+        //@formatter:on
+
+        for (int i = 0; i < size; i++) {
+            Int2 sequential = new Int2(i + (size - i), i + (size - i));
+            assertEquals(sequential.getX(), output.get(i).getX(), 0.001);
+            assertEquals(sequential.getY(), output.get(i).getY(), 0.001);
+        }
+    }
+
+    public static void addVectorInt3(VectorInt3 a, VectorInt3 b, VectorInt3 results) {
+        for (@Parallel int i = 0; i < a.getLength(); i++) {
+            results.set(i, Int3.add(a.get(i), b.get(i)));
+        }
+    }
+
+    @TornadoNotSupported
+    public void testVectorInt3() {
+        int size = 8;
+
+        VectorInt3 a = new VectorInt3(size);
```

Comment:
Why is it not supported?

Reply:
It is supported. 
The latest commit enables all Unit-Test related to Vector Types.